### PR TITLE
Create vultr_lb module

### PR DESCRIPTION
**Pull request by jstoja**
##### SUMMARY

Implement Vultr Load Balancer API (https://www.vultr.com/api/#loadbalancer)

Fixes https://github.com/ansible/ansible/issues/68900

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
vulr_lb

##### ADDITIONAL INFORMATION

Goal is something like this:

```
- vultr_lb:
    dcid: 1
    algorithm: Leastconn
    label: web
    state: present
    forwarding_rules:
      - frontend_protocol: https
        frontend_port: 81
        backend_protocol: https
        backend_port: 81
```

### CODE REVIEW

**Comment by felixfontein** [comment]
You need to add a relative symlink from `plugins/modules/` to your module.

CC @resmo @Spredzy

**Comment by jstoja** [comment]
I've added the symlink now. Please let me know if you have some feedback, this is my first module.

File: plugins/modules/cloud/vultr/vultr_lb.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '0.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: vultr_lb
+short_description: Manages Load Balanders on Vultr.
```
Comment (felixfontein) [inline plugins/modules/cloud/vultr/vultr_lb.py:17]:
```suggestion
short_description: Manages Load Balancers on Vultr
```

File: plugins/modules/cloud/vultr/vultr_lb.py [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+description:
+  - Create and remove Load Balancers.
+author: "Julien BORDELLIER (@jstoja)"
+options:
+  dcid:
+    description:
+      - DCID integer Location in which to create the load balancer.
+    required: true
```
Comment (felixfontein) [inline plugins/modules/cloud/vultr/vultr_lb.py:25]: You need to add `type: xxx` as well. (For **all** options.)

File: plugins/modules/cloud/vultr/vultr_lb.py [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+  forwarding_rules:
+    description:
+      - Defines forwarding rules that your load balancer will follow.
+  ssl_private_key:
+    description:
+      - The SSL certificates private key.
+  ssl_certificate:
+    description:
+      - The SSL Certificate.
+  ssl_chain
```
Comment (felixfontein) [inline plugins/modules/cloud/vultr/vultr_lb.py:56]:
```suggestion
  ssl_chain:
```

File: plugins/modules/cloud/vultr/vultr_lb.py [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+- community.general.vultr
+
+'''
+
+EXAMPLES = r'''
+- name: Ensure a Load Balancer exists
+  local_action:
+    module: vultr_lb
+    dcid: 1
+    algorithm: Leastconn
```
Comment (felixfontein) [inline plugins/modules/cloud/vultr/vultr_lb.py:74]:
```suggestion
    balancing_algorithm: leastconn
```

File: plugins/modules/cloud/vultr/vultr_lb.py [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+    default: present
+    choices: [ present, absent ]
+extends_documentation_fragment:
+- community.general.vultr
+
+'''
+
+EXAMPLES = r'''
+- name: Ensure a Load Balancer exists
+  local_action:
```
Comment (felixfontein) [inline plugins/modules/cloud/vultr/vultr_lb.py:71]:
```suggestion
  community.general.vultr_lb:
```

File: plugins/modules/cloud/vultr/vultr_lb.py [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+    module: vultr_lb
```
Comment (felixfontein) [inline plugins/modules/cloud/vultr/vultr_lb.py:72]:
```suggestion
```

File: plugins/modules/cloud/vultr/vultr_lb.py [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+    dcid: 1
+    algorithm: Leastconn
+    label: web
+    state: present
+    forwarding_rules:
+      - frontend_protocol: https
+        frontend_port: 81
+        backend_protocol: https
+        backend_port: 81
```
Comment (felixfontein) [inline plugins/modules/cloud/vultr/vultr_lb.py:81]:
```suggestion
        backend_port: 81
  delegate_to: localhost
```

File: plugins/modules/cloud/vultr/vultr_lb.py [hunk 8]
```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2020, Julien BORDELLIER <git@julienbordellier.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '0.1',
```
Comment (felixfontein) [inline plugins/modules/cloud/vultr/vultr_lb.py:10]:
```suggestion
ANSIBLE_METADATA = {'metadata_version': '1.1',
```

File: plugins/modules/cloud/vultr/vultr_lb.py [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+                data=data
+            )
+        return lb
+
+
+def main():
+    argument_spec = vultr_argument_spec()
+    argument_spec.update({
+        'name': {
+            'required': True,
```
Comment (felixfontein) [inline plugins/modules/cloud/vultr/vultr_lb.py:267]:
```suggestion
            'type': str,
            'required': True,
```
You need to specify the type for every option.

File: plugins/modules/cloud/vultr/vultr_lb.py [hunk 10]
```diff
@@ -0,0 +1,307 @@
...
+        'state': {
+            'choices': ['present', 'absent'],
+            'default': 'present',
+        },
+    })
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_if=[
+            ('state', 'present', ['name', 'dcid']),
```
Comment (felixfontein) [inline plugins/modules/cloud/vultr/vultr_lb.py:291]: This doesn't make sense, since you marked both options `name` and `dcid` as always required.

**Comment by resmo** [comment]
We actively maintain the vultr modules in the collection https://github.com/ngine-io/ansible-collection-vultr. The PR for cleanup vultr related content in this repo is currently in the making.

**Comment by felixfontein** [comment]
See #172.

**Comment by jstoja** [comment]
No problems :)

Would you like me to open the same PR, but on your repository? Otherwise, I’ll close this one.

**Comment by resmo** [comment]
> No problems :) Would you like me to open the same PR, but on your repository? Otherwise, I’ll close this one.
yes, please create a PR targeting the new repository, feel free to close this PR.

**Comment by jstoja** [comment]
Closing this one in favor of https://github.com/ngine-io/ansible-collection-vultr/pull/3 .

**Pull request closed without merging**